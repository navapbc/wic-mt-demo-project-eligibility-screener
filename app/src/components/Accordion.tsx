import { ReactElement, useState } from 'react'

type Props = {
  body: string
  header: string
}

const Accordion = (props: Props): ReactElement => {
  const { body, header } = props
  const [isExpanded, setExpanded] = useState(false)

  const handleClick = () => {
    setExpanded(!isExpanded)
  }

  return (
    <div className="usa-accordion usa-accordion--bordered">
      <h3 className="usa-accordion__heading">
        <button
          aria-controls="b-a1"
          aria-expanded={isExpanded}
          className="usa-accordion__button"
          onClick={handleClick}
          type="button"
        >
          {header}
        </button>
      </h3>
      <div
        className="usa-accordion__content usa-prose"
        hidden={!isExpanded}
        id="b-a1"
      >
        <p>{body}</p>
      </div>
    </div>
  )
}

export default Accordion
