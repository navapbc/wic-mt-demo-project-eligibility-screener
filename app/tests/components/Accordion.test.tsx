import { render, screen } from '@testing-library/react'

import Accordion from '@components/Accordion'

import { setupUserEvent } from '../helpers/setup'

const testProps = {
  bodyKey: 'body',
  headerKey: 'header',
}

it('should match snapshot', () => {
  const { container } = render(<Accordion {...testProps} />)
  expect(container).toMatchSnapshot()
})

it('should expand the accordion on click', async () => {
  const user = setupUserEvent()
  render(<Accordion {...testProps} />)

  const body = screen.getByText(testProps.bodyKey)
  expect(body).not.toBeVisible()

  const button = screen.getByText(testProps.headerKey)
  await user.click(button)
  expect(body).toBeVisible()
})
