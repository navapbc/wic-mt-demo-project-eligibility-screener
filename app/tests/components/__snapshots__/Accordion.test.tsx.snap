// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="usa-accordion usa-accordion--bordered"
>
  <h3
    className="usa-accordion__heading"
  >
    <button
      aria-controls="b-a1"
      aria-expanded={false}
      className="usa-accordion__button"
      onClick={[Function]}
      type="button"
    >
      header
    </button>
  </h3>
  <div
    className="usa-accordion__content"
    hidden={true}
    id="b-a1"
  >
    body
  </div>
</div>
`;
